### ISSUE TITLE
regression with latest Python 3.11.1 and _socket not found

### ISSUE THREAD

**Issue by endmarsfr** [python/cpython#100171]
Hello, 

I had an error with latest Python 3.11.1 on windows 10 and Windows 11.
`module _socket not found `

it concerns socket.py file line 52

- Everything worked well with previous version 3.11.0
- So i moved back to 3.11.0 version
- it worked again with 3.11.0

I hope, that could be solved easily

Regards

**Comment by ramvikrams** [comment]
Could you please share the code where you got this error

**Comment by pyscripter** [comment]
Same issue here.  I using python embedded to another application (the PyScripter IDE).   It worked fine with 3.11.0 and all previous versions of python.

**Comment by pyscripter** [comment]
Ok I found the source of this issue and a workaround:
 os.path.join(sys.prefix, 'DLLs') is missing from the python path.

The following works.
```python
    dllpath = os.path.join(sys.prefix, 'DLLs')
    sys.path.insert(2, dllpath)
    import _socket
```

Now the question is why  os.path.join(sys.prefix, 'DLLs') is missing...

**Comment by pyscripter** [comment]
https://github.com/python/cpython/commit/3d889dc0a0efa6fcbd984ece365c9ac08d2cd4a9 is culprit. 

```python
# Include DLLs directory as well
paths.insert(1, '.\\DLLs')
```
was for some reason removed.

**Comment by pyscripter** [comment]
https://github.com/python/cpython/commit/3d889dc0a0efa6fcbd984ece365c9ac08d2cd4a9#diff-6fce05f0b43cd31ded3d7c0b3db040a75a61c79874dd660485f040a1773e3418 is probably the culprit.

```
Assumes that a missing ``DLLs`` directory means that standard extension
modules are in the executable's directory.
```

**Comment by endmarsfr** [comment]
Nice job :)

**Comment by zooba** [comment]
I expect this is the same issue as fixed in #100947, since it looks like PyScripter relies on having a separate install of Python. It's not obvious that it's running the executable separately from the original install, but given the timing for the issue appearing I assume it must.

**Comment by pyscripter** [comment]
@zooba Indeed this is the case.  Python is embedded in an executable that it is located in a different directory. 

How can I test whether this resolves the issue?

**Comment by zooba** [comment]
getpath.py is part of the source code. It gets compiled when we build and merged into the DLL so that we can run it before the compiler is ready to run - no easy way to change it without recompiling Python, I'm afraid.

If you set `PYTHONHOME` environment variable to Python's actual install location, it should work.

I think a more accurate workaround is to modify your registry - find the 3.11/PythonPath key and add two new keys into it. In the `(Default)` value for each, put the path to the `Lib` folder in one and the path to `DLLs` in the other. (One issue is that we are only looking at subkeys of this right now, and not the main one - that'll be fixed for 3.11.2 - so putting values in individual subkeys should help them be found. It _will_ mess with your normal install a bit though, probably not badly, but it's an example of why we prefer to have the library in the same location as the executable rather than relying on registry keys.)

**Comment by pyscripter** [comment]
> If you set `PYTHONHOME` environment variable to Python's actual install location, it should work.

I do that for unregistered versions.  For registered versions it used to work fine without it before 3.11.1.   Will it work as before in 3.11.2?

**Comment by zooba** [comment]
It should be fixed for 3.11.2, but I would still recommend setting the variable if you know what it should be set to. The registry values are mostly there for back-compat, and newer installers may not have them (e.g. Windows Store installs do not, Nuget installs do not, etc.)

**Comment by zooba** [comment]
Should be fixed in #100947
